docs/docs-beta/docs/guides/deployment/docker.md

```diff
@@ -1,3 +1,197 @@
+
+To build this Docker image, use a Dockerfile like the following, with a name like `Dockerfile_dagster`:
+
+```dockerfile
+FROM python:3.10-slim
+
+RUN pip install \
+    dagster \
+    dagster-graphql \
+    dagster-webserver \
+    dagster-postgres \
+    dagster-docker
+
+# Set $DAGSTER_HOME and copy dagster.yaml and workspace.yaml there
+ENV DAGSTER_HOME=/opt/dagster/dagster_home/
+
+RUN mkdir -p $DAGSTER_HOME
+
+COPY dagster.yaml workspace.yaml $DAGSTER_HOME
+
+WORKDIR $DAGSTER_HOME
+```
+
+`dagster-postgres` is included to enable using Postgres as your Dagster database. `dagster-docker` is included to enable using the Docker run launcher, which launches each run within a Docker container.
+
+You'll need to have a [workspace file (`workspace.yaml`)](/concepts/code-locations/workspace-files) file in the same directory as the Dockerfile, to tell the web server and daemon where to look for the code servers.
+
+You'll also need to have a [dagster.yaml](https://docs.dagster.io/deployment/dagster-instance) file in the same directory as the Dockerfile, to configure the instance.
+
+### Define a Docker image for each code location
+
+The Docker image for a [code location](/concepts/code-locations) is used for the container for the code server, and for the container for each run launched for that code location.
+
+To build a Docker image for a code location, use a Dockerfile like the following, with a name like `Dockerfile_code_location_1`:
+
+```dockerfile
+FROM python:3.10-slim
+
+RUN pip install \
+    dagster \
+    dagster-postgres \
+    dagster-docker
+
+# Add code location code
+WORKDIR /opt/dagster/app
+COPY directory/with/your/code/ /opt/dagster/app
+
+# Run dagster code server on port 4000
+
+EXPOSE 4000
+
+# CMD allows this to be overridden from run launchers or executors to execute runs and steps
+CMD ["dagster", "code-server", "start", "-h", "0.0.0.0", "-p", "4000", "-f", "definitions.py"]
+```
+
+### Write a docker-compose file for all the services
+
+The following `docker-compose.yaml` defines how to run all the service containers:
+
+```yaml
+version: "3.7"
+
+services:
+  # This service runs the postgres DB used by dagster for run storage, schedule storage,
+  # and event log storage.
+  docker_example_postgresql:
+    image: postgres:11
+    container_name: docker_example_postgresql
+    environment:
+      POSTGRES_USER: "postgres_user"
+      POSTGRES_PASSWORD: "postgres_password"
+      POSTGRES_DB: "postgres_db"
+    networks:
+      - docker_example_network
+
+  # This service runs the code server that loads your user code, in both dagster-webserver
+  # and dagster-daemon. By setting DAGSTER_CURRENT_IMAGE to its own image, we tell the
+  # run launcher to use this same image when launching runs in a new container as well.
+  # Multiple containers like this can be deployed separately - each just needs to run on
+  # its own port, and have its own entry in the workspace.yaml file that's loaded by the
+      # webserver.
```

Comment:
Curious what @PedramNavid and @cmpadden think about including a block of text this big inside code. 

Context is good, but:

- Having to scroll sideways to read it was kind of annoying
- Should this be included in a step in another section?

Reply:
I agree that the sideways scrolling is not ideal.

Having it outside the code block has the added benefit that users can copy and paste the code more easily.

If we have a more advanced section of docs explaining environment variables like `DAGSTER_CURRENT_IMAGE`, then we can link to that and keep the guide more minimal.

Reply:
I tend to agree, if its outside the code block we'll also get spell-check / linting.

Reply:
Oops this just came in from the example I based this on. I don't think we need it.